receiver/prometheusreceiver/config.go

```diff
@@ -47,93 +36,77 @@ type Config struct {
 	// ReportExtraScrapeMetrics - enables reporting of additional metrics for Prometheus client like scrape_body_size_bytes
 	ReportExtraScrapeMetrics bool `mapstructure:"report_extra_scrape_metrics"`
 
-	TargetAllocator *targetAllocator `mapstructure:"target_allocator"`
-
-	// ConfigPlaceholder is just an entry to make the configuration pass a check
-	// that requires that all keys present in the config actually exist on the
-	// structure, ie.: it will error if an unknown key is present.
-	ConfigPlaceholder any `mapstructure:"config"`
+	TargetAllocator *TargetAllocator `mapstructure:"target_allocator"`
 
 	// EnableProtobufNegotiation allows the collector to set the scraper option for
 	// protobuf negotiation when conferring with a prometheus client.
 	EnableProtobufNegotiation bool `mapstructure:"enable_protobuf_negotiation"`
 }
 
-type targetAllocator struct {
+// Validate checks the receiver configuration is valid.
+func (cfg *Config) Validate() error {
+	if cfg.PrometheusConfig != nil && len(cfg.PrometheusConfig.ScrapeConfigs) == 0 && cfg.TargetAllocator == nil {
```

Comment:
Not sure I agree with your change, because that will be a behavior change (which I am avoiding in a refactor PR), here is the previous code:

```golang
func (cfg *Config) Validate() error {
	promConfig := cfg.PrometheusConfig
	if promConfig != nil {
		err := cfg.validatePromConfig(promConfig)
		if err != nil {
			return err
		}
	}
	...
}

func (cfg *Config) validatePromConfig(promConfig *promconfig.Config) error {
	if len(promConfig.ScrapeConfigs) == 0 && cfg.TargetAllocator == nil {
		return errors.New("no Prometheus scrape_configs or target_allocator set")
	}
	....
}
```

You can see that if the  `promConfig` is `nil` will not be an error in the previous code. I understand that my refactored code makes it more obvious that this may be a bug, but we can have a fix in a followup PR.

Reply:
If you won't fix this in this PR then I want to see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30135 merged first.

Reply:
This bug was here for more than an year, and you are only causing a lot of issues to rebase this PR. I think it is not ok to push for this.

Reply:
PR merged first. Closing this, please let me know what else you want to see to unblock this PR.